hack/check-docforge.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+echo "> Check Manifest"
+
+if [[ $(uname) == 'Darwin' ]]; then
+  READLINK_BIN="greadlink"
+else
+  READLINK_BIN="readlink"
+fi
+
+repoPath="$(${READLINK_BIN} -f $(dirname ${0})/..)"
+manifestPath="${repoPath}/.docforge/manifest.yaml"
+diffDirs=".docforge/;docs/"
+
+tmpDir=$(mktemp -d)
+function cleanup {
+    rm -rf "$tmpDir"
+}
+trap cleanup EXIT ERR INT TERM
+
+curl https://raw.githubusercontent.com/gardener/documentation/master/.ci/check-manifest --output ${tmpDir}/check-manifest-script.sh && chmod +x ${tmpDir}/check-manifest-script.sh
+curl https://raw.githubusercontent.com/gardener/documentation/master/.ci/check-manifest-config --output ${tmpDir}/manifest-config
+scriptPath="$(${READLINK_BIN} -f ${tmpDir}/check-manifest-script.sh)"
+configPath="$(${READLINK_BIN} -f ${tmpDir}/manifest-config)"
```

Comment:
Instead of downloading the scripts each and every time, can we rather vendor them?

You'd need to define a `doc.go` file in the https://github.com/gardener/documentation/tree/master/.ci directory, similar to https://github.com/gardener/gardener/blob/master/hack/.ci/doc.go or https://github.com/kubernetes/code-generator/blob/master/tools.go. Then others (like g/g or extension) can vendor the script without the necessity to download it each and every time.
This is also what we do such that extensions can reuse the `hack` scripts of g/g, see e.g. https://github.com/gardener/gardener-extension-provider-aws/blob/74076885432d7f1c29462c772065a5596b8df57d/Makefile#L128

Following this approach most likely allows to drop most of the boilerplate above, potentially even the entire script (it could maybe all be done in the `Makefile`)?!

Reply:
Hmm isn't it better approach if we just attach the tools directory with the binaries as a CI/CD persistent disk. 
This can be done with `cache-path` configuration: https://gardener.github.io/cc-utils/pipeline_step.html?highlight=cache_paths#pipeline-step.
This cache can be added to the `verify` step

Reply:
This is also contradicting to https://github.com/gardener/gardener/pull/5036#issuecomment-972819332, isn't it?

Reply:
I have mentioned the above approach because this check will be added to all repos with manifest file and it will be better if we keep the consistency between them. As some of them like `dashboard` for example are not `golang` project, your approach will not be possible for them

Reply:
OK, I synced with Tim and we are fine with this approach, however, we would like you do download a specific version of the script (not always the `master` version) such that we achieve reproducability and get a bit more robust to flakes.

Reply:
OK, are you fine if we use a hash of commit because we do not have releases of the documentation repo.

Reply:
Sure 👍🏻